MagazineLayout/Public/MagazineLayout.swift

```diff
@@ -586,11 +586,26 @@ public final class MagazineLayout: UICollectionViewLayout {
   }
 
   override public func shouldInvalidateLayout(forBoundsChange newBounds: CGRect) -> Bool {
-    let isSameWidth = collectionView?.bounds.size.width.isEqual(
-      to: newBounds.size.width,
-      threshold: 1 / scale)
-      ?? false
-    return !isSameWidth || hasPinnedHeaderOrFooter
+    // When using the topToBottom layout direction, we only want
+    // to invalidate the layout when the widths differ.
+    // When using the bottomToTop layout direction, we want to
+    // invalidate on any size change due to the requirement of
+    // needing to preserve scroll position from the bottom
+    let shouldInvalidateDueToSize = switch verticalLayoutDirection {
+    case .topToBottom:
+      !currentCollectionView.bounds.size.width.isEqual(
+        to: newBounds.size.width,
+        threshold: 1 / scale)
+    case .bottomToTop:
+      !(currentCollectionView.bounds.size.width.isEqual(
+        to: newBounds.size.width,
+        threshold: 1 / scale) &&
+      currentCollectionView.bounds.size.height.isEqual(
+        to: newBounds.size.height,
+        threshold: 1 / scale))
+    }
```

Comment:
because we support back to iOS 11 (we can probably change this in the future), we can't use this fancy new syntax.